core/src/avm2/activation.rs

```diff
@@ -1232,9 +1241,26 @@ impl<'a, 'gc, 'gc_context> Activation<'a, 'gc, 'gc_context> {
         method: Gc<'gc, BytecodeMethod<'gc>>,
         index: Index<AbcMultiname>,
     ) -> Result<FrameControl<'gc>, Error> {
-        let multiname = self.pool_multiname(method, index)?;
+        let lookupname = self.pool_lookupname(method, index)?;
         let object = self.context.avm2.pop().coerce_to_object(self)?;
 
+        if let Some(dictionary) = object.as_dictionary_object() {
+            if lookupname
+                .local_value()
+                .map(|v| !v.is_boxed_primitive())
```

Comment:
hmm, when can a stack value be a boxed primitive? I thought PrimitiveObject is there just to have values-as-objects for some opcodes, right? And if it can get leaked, maybe we should guarantee that it gets un-boxed before arriving back on stack?

(and now that I look at it, I feel like many uses of `.coerce_as_object()` should instead be a fallible `.as_object()?`)

Reply:
AFAIK, stack values aren't restricted at any point to being unboxed primitives, but I'm also struggling to recall a code path that would let you actually store a boxed primitive anywhere. I added `is_boxed_primitive` just in case there's a boxed object leak.

It probably would make sense to automatically unbox primitives instead. AFAIK, the object shell is just to let `callproperty` work, so it's semantically equivalent to the bare value. I'll probably want to PR that separately, though.

Reply:
>  I'll probably want to PR that separately, though.

Sure :)

> AFAIK, the object shell is just to let callproperty work

Now that I think about it, after we get around to the PR that'll move instance traits to `ClassObject`/`Traits`/`VTable`/whatever, maybe it would be feasible to not "box" the primitives and all and directly lookup on their `ClassObject`/`Traits`/`VTable`/whatever, as in something like:

```rs
let cls = match Value {
    Bool(_) => avm2.classes().bool,
    String(_) => avm2.classes().string,
    // etc
    Object(obj) => obj.instance_of(), // or: obj.vtable() etc
}
let ... = cls.resolve(multiname);
// ...
```

Reply:
Unfortunately those methods have a signature that requires an `Object` receiver, so we'd still need to box the primitive if we wanted to call them.